client/components/date-range/index.js

```diff
@@ -0,0 +1,526 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { bindAll, noop, isNil } from 'lodash';
+import { DateUtils } from 'react-day-picker';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { localize, moment } from 'i18n-calypso';
+import DatePicker from 'components/date-picker';
+import Popover from 'components/popover';
+import DateRangeInputs from './inputs';
+import DateRangeHeader from './header';
+import DateRangeTrigger from './trigger';
+
+export class DateRange extends Component {
+	static propTypes = {
+		selectedStartDate: PropTypes.oneOfType( [
+			PropTypes.instanceOf( Date ),
+			PropTypes.instanceOf( moment ),
+		] ),
+		selectedEndDate: PropTypes.oneOfType( [
+			PropTypes.instanceOf( Date ),
+			PropTypes.instanceOf( moment ),
+		] ),
+		onDateSelect: PropTypes.func,
+		onDateCommit: PropTypes.func,
+		firstSelectableDate: PropTypes.oneOfType( [
+			PropTypes.instanceOf( Date ),
+			PropTypes.instanceOf( moment ),
+		] ),
+		lastSelectableDate: PropTypes.oneOfType( [
+			PropTypes.instanceOf( Date ),
+			PropTypes.instanceOf( moment ),
+		] ),
+		triggerText: PropTypes.func,
+		renderTrigger: PropTypes.func,
+		renderHeader: PropTypes.func,
+		renderInputs: PropTypes.func,
+	};
+
+	static defaultProps = {
+		onDateSelect: noop,
+		onDateCommit: noop,
+		renderTrigger: props => <DateRangeTrigger { ...props } />,
+		renderHeader: props => <DateRangeHeader { ...props } />,
+		renderInputs: props => <DateRangeInputs { ...props } />,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		// Define the date range as Moment instances
+		const firstSelectableDate = ! isNil( this.props.firstSelectableDate )
+			? this.props.moment( this.props.firstSelectableDate )
+			: undefined;
+		const lastSelectableDate = ! isNil( this.props.lastSelectableDate )
+			? this.props.moment( this.props.lastSelectableDate )
+			: undefined;
+
+		// Clamp dates to ranges (if specified)
+		let startDate;
+		let endDate;
+
+		endDate = ! isNil( this.props.selectedEndDate )
```

Comment:
We can invert the ternaries here to avoid having to negate the checks:

```js
endDate = isNil( this.props.selectedEndDate )
  ? this.props.moment()
  : this.props.moment( this.props.selectedEndDate );
```

Reply:
Yeh that is clearer - thanks.

Reply:
Done